connect/src/test/java/com/stripe/android/connect/webview/StripeConnectWebViewContainerControllerTest.kt

```diff
@@ -36,4 +47,54 @@ class StripeConnectWebViewContainerControllerTest {
             "https://connect-js.stripe.com/v1.0/android_webview.html#component=payouts&publicKey=pk_test_123"
         )
     }
+
+    @Test
+    fun `shouldOverrideUrlLoading doesn't handle request when request is null, returns false`() {
+        val resultNullRequest = controller.shouldOverrideUrlLoading(context = mockContext, request = null)
+        assertFalse(resultNullRequest)
+    }
+
+    @Test
+    fun `shouldOverrideUrlLoading allows request and returns false for allowlisted hosts`() {
+        val uri = mockUri("https", "connect-js.stripe.com", "/allowlisted")
+        val mockRequest = mock<WebResourceRequest> {
+            on { url } doReturn uri
+        }
+
+        val result = controller.shouldOverrideUrlLoading(mockContext, mockRequest)
+        assertFalse(result)
+    }
+
+    @Test
+    fun `shouldOverrideUrlLoading launches ChromeCustomTab for https urls`() {
+        val mockUri = mockUri("https", "example.com", "/test")
+        val mockRequest = mock<WebResourceRequest> {
+            on { url } doReturn mockUri
+        }
+
+        val result = controller.shouldOverrideUrlLoading(mockContext, mockRequest)
+
+        assert(result)
+        verify(mockStripeIntentLauncher).launchSecureExternalWebTab(mockContext, mockUri)
+    }
+
+    @Test
+    fun `shouldOverrideUrlLoading blocks url loading and returns true for unsupported schemes`() {
+        val mockUri = mockUri("mailto", "example.com", "/email")
+        val mockRequest = mock<WebResourceRequest> {
+            on { url } doReturn mockUri
+        }
+
+        val result = controller.shouldOverrideUrlLoading(mockContext, mockRequest)
+        assert(result)
+    }
+
+    private fun mockUri(scheme: String, host: String, path: String): Uri {
+        return mock<Uri> {
```

Comment:
Can we build a real `Uri` instance instead of mock? May need to use Robolectric for Android APIs, which should be OK?

Reply:
Yeah we need to use Robolectric, which is why I went with this (I don't love how slow Robolectric makes things). I don't feel super strongly, 👍 to this approach